client/me/account-password/index.jsx

```diff
@@ -98,42 +98,14 @@ const AccountPassword = createReactClass( {
 		this.props.recordGoogleEvent( 'Me', 'Focused on New Password Field' );
 	},
 
-	submitForm: function ( event ) {
-		const { translate, errorNotice: showErrorNotice } = this.props;
-
+	submitForm( event ) {
 		event.preventDefault();
 
-		this.setState( {
-			savingPassword: true,
-		} );
-
-		this.props.userSettings.saveSettings(
-			function ( error, response ) {
-				this.setState( {
-					savingPassword: false,
-					isUnsaved: false,
-				} );
-
-				if ( error ) {
-					debug( 'Error saving password: ' + JSON.stringify( error ) );
-
-					// handle error case here
-					showErrorNotice(
-						translate( 'There was a problem saving your password. Please, try again.' )
-					);
-					this.setState( { submittingForm: false } );
-				} else {
-					debug( 'Password saved successfully' + JSON.stringify( response ) );
-
-					// Since changing a user's password invalidates the session, we reload.
-					window.location = window.location.pathname + '?updated=password';
-				}
-			}.bind( this ),
-			{ password: this.state.password }
-		);
+		this.setState( { isUnsaved: false } );
```

Comment:
Note: this should happen after the request for a password change.

Reply:
Yeah, I can see a difference in the behavior in prod vs this branch here. To repro, you'll need to set your network to slow 3g or another means of slow internet connection:

* In production, if you submit the form to change your password, while submitting if you click the "Back" button, you get the alert that asks if you're sure you want to abandon unsaved data.
* In this branch, when you click the "Back" button while submitting, you just go to the main `/me/security` page without any alert.

Reply:
I've got an experimental fix here: 813b44dd80f0aaa92b40c30b01ec82063128e45a – let me know what you think. Calling `setState` in `cDU` is discouraged, because it causes another rerender. However, that might be what we want anyway so using it conditionally might be okay?